### PR TITLE
Ticdc(Sink) use ticdc logger to print pulsar log

### PR DESCRIPTION
**Pull request by yumchina**
2 set partitionKey value to message

### What problem does this PR solve?

1 use ticdc logger to print pulsar log
2 set partitionKey value to message
Issue Number: ref https://github.com/pingcap/tiflow/issues/9413

#### Tests 

 - Unit test
 

### Release note 
```release-note
None
```

### CODE REVIEW

File: pkg/sink/pulsar/factory.go [hunk 1]
```diff
@@ -93,7 +95,10 @@ func setupAuthentication(config *config.PulsarConfig) (pulsar.Authentication, er
 	if config.AuthTLSCertificatePath != nil && config.AuthTLSPrivateKeyPath != nil {
 		return pulsar.NewAuthenticationTLS(*config.AuthTLSCertificatePath, *config.AuthTLSPrivateKeyPath), nil
 	}
-	return nil, fmt.Errorf("no authentication method found")
+
+	log.Warn("pulsar client no authentication method found")
+
+	return nil, nil
```
Comment (3AceShowHand) [inline pkg/sink/pulsar/factory.go:101]: should return error here ?

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/sink/pulsar/factory.go (hunk 1 above)
Comment (asddongmen) [reply pkg/sink/pulsar/factory.go:101]: Don't need an error, nil means no credentials are provided.

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/pulsar/factory.go (hunk 1 above)
Comment (yumchina) [reply pkg/sink/pulsar/factory.go:101]: No, pulsar supports to create a client without any Authentications

File: cdc/sink/dmlsink/mq/dispatcher/partition/dispatcher.go [hunk 2]
```diff
@@ -21,5 +21,5 @@ import (
 type Dispatcher interface {
 	// DispatchRowChangedEvent returns an index of partitions according to RowChangedEvent.
 	// Concurrency Note: This method is thread-safe.
-	DispatchRowChangedEvent(row *model.RowChangedEvent, partitionNum int32) int32
+	DispatchRowChangedEvent(row *model.RowChangedEvent, partitionNum int32) (int32, *string)
```
Comment (sdojjy) [inline cdc/sink/dmlsink/mq/dispatcher/partition/dispatcher.go:24]: update the function comment

File: cdc/sink/dmlsink/mq/dispatcher/partition/index_value.go [hunk 3]
```diff
@@ -56,5 +56,6 @@ func (r *IndexValueDispatcher) DispatchRowChangedEvent(row *model.RowChangedEven
 			r.hasher.Write([]byte(col.Name), []byte(model.ColumnValueString(col.Value)))
 		}
 	}
-	return int32(r.hasher.Sum32() % uint32(partitionNum))
+	sum32 := r.hasher.Sum32()
+	return int32(sum32 % uint32(partitionNum)), num2str(sum32)
```
Comment (asddongmen) [inline cdc/sink/dmlsink/mq/dispatcher/partition/index_value.go:60]: Why do not simply use `row.Table.String()`  as the partition key?

File: cdc/sink/dmlsink/mq/dispatcher/partition/ts.go [hunk 4]
```diff
@@ -25,6 +30,6 @@ func NewTsDispatcher() *TsDispatcher {
 
 // DispatchRowChangedEvent returns the target partition to which
 // a row changed event should be dispatched.
-func (t *TsDispatcher) DispatchRowChangedEvent(row *model.RowChangedEvent, partitionNum int32) int32 {
-	return int32(row.CommitTs % uint64(partitionNum))
+func (t *TsDispatcher) DispatchRowChangedEvent(row *model.RowChangedEvent, partitionNum int32) (int32, *string) {
+	return int32(row.CommitTs % uint64(partitionNum)), aws.String(fmt.Sprintf("%d", row.CommitTs))
```
Comment (sdojjy) [inline cdc/sink/dmlsink/mq/dispatcher/partition/ts.go:34]: seems we have a function named num2str, can we reuse it?

File: cdc/sink/dmlsink/mq/dispatcher/event_router.go [hunk 5]
```diff
@@ -226,13 +232,17 @@ func (s *EventRouter) matchDispatcher(
...
-	ruleConfig *config.DispatchRule, enableOldValue bool,
+	ruleConfig *config.DispatchRule, enableOldValue bool, schema string,
 ) partition.Dispatcher {
 	var (
 		d    partition.Dispatcher
 		rule partitionDispatchRule
 	)
-	rule.fromString(ruleConfig.PartitionRule)
+	notInDispatchRule := rule.fromString(ruleConfig.PartitionRule)
+	if schema == sink.PulsarScheme && notInDispatchRule { // in pulsar,if not in rule values, transport the value
```
Comment (asddongmen) [inline cdc/sink/dmlsink/mq/dispatcher/event_router.go:242]: should also add PulsarSSLSchema

File: cdc/sink/dmlsink/mq/dispatcher/partition/dispatcher.go (hunk 2 above)
Comment (3AceShowHand) [reply cdc/sink/dmlsink/mq/dispatcher/partition/dispatcher.go:24]: What is the usage of the second returned value, it's set as `num2str(sum32)`

**Review by 3AceShowHand — APPROVED** [review]

**Comment by asddongmen** [comment]
This pull request was split into #9674 #9677.

**Pull request closed without merging**